### ISSUE TITLE
I18nContext not available in child component in component library with webpack5

### ISSUE THREAD

**Issue by skumar5-bech** [lingui/js-lingui#1500]
**Describe the bug**

I18nContext not available in child component in component library with webpack5 and throwing below error :
**Uncaught Error: useLingui hook was used without I18nProvider.**

**To Reproduce**
Code sample is available at this path where this issue is replicable
https://github.com/skumar5-bech/lingui-webpack5
To run the code, follow READ ME.

The sample is created using yarn workspace and has following:

1) React Application (lingui-sample) is created with create-react-app template
2) A reusable component library (common) is created with webpack5

Application & Components Illustration is given below to understand the issue:

React App ( (lingui-sample)
App Component -- App.tsx
----&lt;i18nProvider i18n={i18n}&gt;
---------&lt;Text&gt;&lt;/Text&gt; Referenced from reusable component library (common) 
-----&lt;18nProvider /&gt;

Reusable component library (common)
Text Component -- text/index.tsx
when trying to access i18nContext in this component using below code snippet then this issue will occur.
 const { i18n } = useLingui();

**Expected behavior**
I18nContext should be available to child components if parent is wrapped with i18nProvider.

**Additional context**

Other context providers (ThemeProvider from styled-components and custom) are working with webpack5 and also implemented in code sample provided in reproducible steps.

**Comment by andrii-bodnar** [comment]
Hi @skumar5-bech,

currently, we have a v4 pre-release that contains a lot of improvements, fixes, and other changes. Could you, please check if the issue is reproducible on the Lingui `v4.0.0-next.1`?

refs: [v4.0.0-next.0](https://github.com/lingui/js-lingui/releases/tag/v4.0.0-next.0), [v4.0.0-next.1](https://github.com/lingui/js-lingui/releases/tag/v4.0.0-next.1)

**Comment by thekip** [comment]
@andrii-bodnar this will not help for the issue. @skumar5-bech but you still give a try because there are many improvements and it quite stable 

@skumar5-bech, the problem is that your application and library consume different versions of `@lingui` dependencies. Application consumes esm version and library consumes cjs. Hence, you have two copies of the lib in the bundle, and 
 therefore react's context doesn't work. 

That happens because you library is built with cjs modules and have externalized imports like this: 

```ts
/***/ "@lingui/react":
/*!********************************!*\
  !*** external "@lingui/react" ***!
  \********************************/
/***/ ((module) => {

module.exports = require("@lingui/react");

/***/ }),
``` 

And this require according to `@lingui/react`'s package.json: 

```json
  "exports": {
    ".": {
      "require": {
        "types": "./build/index.d.ts",
        "default": "./build/cjs/index.js"
      },
      "import": {
        "types": "./build/index.d.ts",
        "default": "./build/esm/index.js"
      }
    },
    "./package.json": "./package.json"
  },
```
will be resolved to `./build/cjs/index.js"` 

Where your host app will consume `./build/esm/index.js` because by default bundlers take `import` conditional export. 
```

You need to setup your library to build in ESM format or in dual.

**Comment by skumar5-bech** [comment]
For above scenario, I18nContext is available with 3.14.0 version but with version > 3.14.0, it's an issue. Why is this so?

**Comment by thekip** [comment]
Because previously it was distributed as just commonjs package, now it's distributed as dual package.
